cmd/restic/cmd_check.go

```diff
@@ -76,6 +85,17 @@ func checkFlags(opts CheckOptions) error {
 			if dataSubset[1] > totalBucketsMax {
 				return errors.Fatalf("check flag --read-data-subset=n/t t must be at most %d", totalBucketsMax)
 			}
+		} else if isFileSizeSuffix(opts.ReadDataSubset) {
```

Comment:
I'd prefer to check whether `ReadDataSubset` is a percentage by looking for whether the parameter ends with `%`.